config/routes.rb

```diff
@@ -84,11 +84,19 @@
         resource :preview, only: [:update], controller: 'post_previews'
         member do
           get 'quote'
-          post 'mark_as_unread'
+          post 'mark_as_unread' # deprecated
         end
       end
     end
   end
 
+  # flat urls for anything which is non-visible to users & search engines
+  resources :posts, only: %i[] do
```

Comment:
Alternatively, we could have this block under `resources :topics` but with `shallow: true` (https://edgeguides.rubyonrails.org/routing.html#shallow-nesting).

Also, we could scope all such routes under e.g. `/action/` so that they are easy to block with `robots.txt`.

Reply:
I think it's a little easier to read and rationalize about without shallow, but this might be just a question of taste (I find our routes file rather heavy reading presently).

None of these routes will be `get` I think, so is the robots.txt blocking `/action` prefix really needed?

Reply:
(thinking more). But there might be some clarity to having all of these routes under `action/` (in terms of logging, but also logical separation from the routes that can generate 404s) -- would then need add it to list of reserved words for messageboards (https://github.com/thredded/thredded/blob/master/app/models/thredded/messageboard.rb#L9).

Reply:
> I think it's a little easier to read and rationalize about without shallow, but this might be just a question of taste (I find our routes file rather heavy reading presently).

I don't mind either way. If we want to scope them under `action/`, they'll have to be separate anyway.

> would then need add it to list of reserved words for messageboards 

Yes.